tools-local/Microsoft.ML.InternalCodeAnalyzer/NameFixProvider.cs

```diff
@@ -49,9 +48,7 @@ public override FixAllProvider GetFixAllProvider()
 
         public override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            var diagnostic = context.Diagnostics.FirstOrDefault(d => _fixableSet.Contains(d.Id));
-            if (diagnostic == null)
-                return;
+            var diagnostic = context.Diagnostics[0];
```

Comment:
Are you sure this change is functionally the same? Looks like before it wouldn't always  return a diagnostic where as now it always will.

Reply:
roslyn guarantees at least one diagnostic